src/dom/dom.js

```diff
@@ -19,64 +19,45 @@
 import p5 from '../core/main';
 
 /**
- * Searches the page for an element with the given ID, class, or tag name (using the '#' or '.'
- * prefixes to specify an ID or class respectively, and none for a tag) and returns it as
- * a <a href="#/p5.Element">p5.Element</a>. If a class or tag name is given with more than 1 element,
- * only the first element will be returned.
+ * Searches the page for the first element that matches the given CSS selector string (can be an
+ * ID, class, tag name or a combination) and returns it as a <a href="#/p5.Element">p5.Element</a>.
  * The DOM node itself can be accessed with .elt.
  * Returns null if none found. You can also specify a container to search within.
  *
  * @method select
- * @param  {String} name id, class, or tag name of element to search for
- * @param  {String|p5.Element|HTMLElement} [container] id, <a href="#/p5.Element">p5.Element</a>, or
+ * @param  {String} selectors CSS selector string of element to search for
+ * @param  {String|p5.Element|HTMLElement} [container] CSS selector string, <a href="#/p5.Element">p5.Element</a>, or
  *                                             HTML element to search within
  * @return {p5.Element|null} <a href="#/p5.Element">p5.Element</a> containing node found
  * @example
- * <div ><code class='norender'>
+ * <div><code>
  * function setup() {
- *   createCanvas(100, 100);
- *   //translates canvas 50px down
- *   select('canvas').position(100, 100);
+ *   createCanvas(50, 50);
+ *   background(30);
+ *   // move canvas down and right
+ *   select('canvas').position(10, 30);
  * }
  * </code></div>
- * <div><code class='norender'>
- * // these are all valid calls to select()
- * let a = select('#moo');
- * let b = select('#blah', '#myContainer');
- * let c, e;
- * if (b) {
- *   c = select('#foo', b);
- * }
- * let d = document.getElementById('beep');
- * if (d) {
- *   e = select('p', d);
+ *
+ * <div class="norender"><code>
+ * // select using ID
+ * let a = select('#container');
+ * let b = select('#beep', '#container');
+ * let c;
+ * if (a) {
+ *   // select using class
+ *   c = select('.boop', a);
  * }
+ * // select using CSS selector string
+ * let d = select('#container #bleep');
+ * let e = select('#container p');
  * [a, b, c, d, e]; // unused
  * </code></div>
  */
 p5.prototype.select = function(e, p) {
   p5._validateParameters('select', arguments);
-  var res = null;
   var container = getContainer(p);
-  if (e[0] === '.') {
-    e = e.slice(1);
-    res = container.getElementsByClassName(e);
-    if (res.length) {
-      res = res[0];
-    } else {
-      res = null;
-    }
-  } else if (e[0] === '#') {
-    e = e.slice(1);
-    res = container.getElementById(e);
-  } else {
-    res = container.getElementsByTagName(e);
-    if (res.length) {
-      res = res[0];
-    } else {
-      res = null;
-    }
-  }
+  var res = container.querySelector(e);
```

Comment:
these should be `const` now that we've switched to ES6, not sure how they were missed before